client/build.rs

```diff
@@ -15,7 +15,7 @@ fn main() -> Result<()> {
     // HACK: used by Nix, since at the moment
     // the checks are a process that's hard to accomodate
     // in Nix environment
-    if std::option_env!("IROHA_SKIP_WASM_CHECKS").is_none() {
+    if std::option_env!("IROHA_SKIP_WASM_CHECKS").is_some() {
         check_all_smartcontracts()?;
     }
```

Comment:
is it possible to run this check only if run with `dev` profile?

Reply:
`PROFILE` env variable is set for build scripts: https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts, so it is technically possible

Reply:
what about `cfg(test)`? that is probably not accessible.

Because in fact we would only want smart_contracts built when testing

Reply:
Yeah, I think `cfg(test)` wouldn't work in a `build.rs`

But it can be made into an integration test, which seems to be closer to what we actually want?

Reply:
> But it can be made into an integration test, which seems to be closer to what we actually want?

can you explain what you mean?

Reply:
> can you explain what you mean?

remove client's `build.rs`, put the following into `client/tests/check_all_smart_contracts.rs`:

```rust
use eyre::Result;

const TEST_SMARTCONTRACTS_DIR: &str = "tests/integration/smartcontracts";

#[test]
fn check_all_smartcontracts() -> Result<()> {
    iroha_wasm_builder::Builder::new(TEST_SMARTCONTRACTS_DIR)
        .format()
        .check()
}
```

This will invoke `cargo check` on smart contracts as a separate test, which seems to be actually close to what we want to do (check it when testing).

Reply:
Yeah, I would agree, it should something done deliberately

Reply:
![image](https://github.com/hyperledger/iroha/assets/40506647/79115e79-b0ab-4629-91bb-2f64878bb81a)
 I could see there was a PROFILE variable for test as shown in the picutre above or did I get something wrong here?

Reply:
> I could see there was a PROFILE variable for test as shown in the picutre above or did I get something wrong here?

The `PROFILE` will indeed be passed to the `build.rs` script, but it would have a value of `test`, it will be `debug`, `release` or another value in case of custom profiles.

Profiles are independent from whether the crate is being built in test mode (what `#[cfg(test)]` ultimately checks).

It's actually by design that there's no way to detect whether crate is being built in test mode in `build.rs`: https://github.com/rust-lang/cargo/issues/4789

Reply:
Look [here](https://doc.rust-lang.org/cargo/reference/profiles.html). There is a `test` profile

Reply:
UPD: apparently I was wrong, there _is_ a profile called `test`. However it's a bit weird: it seems that it's only used for an actual test executable (see https://github.com/rust-lang/cargo/issues/8904#issuecomment-734881177) and does not get passed to `build.rs` according to my testing